src/client/GameWindow.cpp

```diff
@@ -80,9 +80,52 @@ GameWindow::GameWindow() {
                 switch (event.type)
                 {
                 case sf::Event::MouseButtonPressed:
+            
+                    clickStartingPoint = {  sf::Mouse::getPosition(clientGameWindow).x, 
+                                            sf::Mouse::getPosition(clientGameWindow).y};
+
+                    if (!mooveMode) {
+
+                        int minimumDistance = WINDOW_LENGTH;
+                        std::array<int, 2> hexagonOnClick = {0, 0};
+                        std::array<int, 2> firstHexagon = {WINDOW_LENGTH, WINDOW_WIDTH};
+
+                        sf::Rect cursorRect = mapTextureToDisplay[0].getSprite(0).getGlobalBounds();
+                        cursorRect.left = clickStartingPoint[0];
+                        cursorRect.top = clickStartingPoint[1];
+                        cursorRect.width = 1;
+                        cursorRect.height = 1;
+
+                        bool isClickable = false;
+
+                        for(int i = 0; i < NUMBER_OF_FIELD; i++){
+
+                            for(unsigned j = 0; j < mapTextureToDisplay[i].getSize(); j++){
+
+                                sf::Rect spriteBounds = mapTextureToDisplay[i].getSprite(j).getGlobalBounds();
+
+                                if (firstHexagon[0] > spriteBounds.left) firstHexagon[0] = spriteBounds.left;
+                                if (firstHexagon[1] > spriteBounds.top) firstHexagon[1] = spriteBounds.top;
+
+                                if (spriteBounds.intersects(cursorRect)) {
+
+                                    isClickable = true;
+
+                                    int distance = pow( pow(spriteBounds.left + spriteBounds.width/2 - cursorRect.left, 2) 
+                                                    + pow(spriteBounds.top + spriteBounds.height/2 - cursorRect.top, 2), 0.5);
```

Comment:
You can use "sqrt" function.
It can also be useful to create a "calcul distance" function.

Reply:
Done